[PATCH] n1sdp: Enabling CoreSight debug non-secure access

Upstream-Status: Pending [Not submitted to upstream yet]

Signed-off-by: Khasim Syed Mohammed <khasim.mohammed@arm.com>

From 194e6239737272c3fadb202deb8015cb6e8df6ec Mon Sep 17 00:00:00 2001
From: Khasim Mohammed <khasim.mohammed@arm.com>
Date: Thu, 22 Apr 2021 15:48:02 +0100
Subject: [PATCH] n1sdp: Enabling CoreSight debug non-secure access

This patch enables coresight to work with n1sdp c2c profile.

Signed-off-by: Khasim Syed Mohammed <khasim.mohammed@arm.com>
---
 product/n1sdp/module/n1sdp_system/src/mod_n1sdp_system.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/product/n1sdp/module/n1sdp_system/src/mod_n1sdp_system.c b/product/n1sdp/module/n1sdp_system/src/mod_n1sdp_system.c
index 70e2ec3f..ff380ae2 100644
--- a/product/n1sdp/module/n1sdp_system/src/mod_n1sdp_system.c
+++ b/product/n1sdp/module/n1sdp_system/src/mod_n1sdp_system.c
@@ -466,6 +466,12 @@ static int n1sdp_system_init_primary_core(void)
             return status;
     }
 
+    else {
+        /* Enable non-secure CoreSight debug access */
+        FWK_LOG_INFO("N1SDP SYSTEM] Enabling CoreSight debug non-secure access");
+        *(volatile uint32_t *)(AP_SCP_SRAM_OFFSET +
+                               NIC_400_SEC_0_CSAPBM_OFFSET) = 0xFFFFFFFF;
+    }
     return FWK_SUCCESS;
 }
 
-- 
2.17.1

